### ISSUE TITLE
Using external audio files for oscillators?

### ISSUE THREAD

**Issue by unfa** [zynaddsubfx/zyn-fusion-issues#136]
Would it be possible to load external audio files as oscillator buffers?

The most obvious use would be  using that for ADDsynth, but since PADsynth also uses a unified oscillator interface, it might make sense to have that available there too.

It would be nice to have a custom audio file available as a source function in the osciallator, just like a Saw↔Tri function etc.

That's by no means a priority, but I wonder if it's possible to do in the current architecture, and if LV2 allows stuff like that. Fabla2 suggest to me it probably does (it an even capture audio).

**Comment by fundamental** [comment]
Possible, yes. To a degree I could see some of this functionality being useful in importing external information. However I will say that there's a line here. Zyn is fundamentally a *synthesizer* and not a sampler. In other words tools which make it easier to control/setup the synthesizer are in scope and loading samples/playing back samples is out-of-scope.

**Comment by unfa** [comment]
Yes, I don't mean this as a sampler functionality. I guess sometimes it could be very useful to be able to use a short audio clip for oscillator or modulation source. Especailly if one can still apply all the crazy distortion, modulation and filtering in the oscillator module as usual.

Being able to have sample playback *cough* _Padsynth_ *cough* underneath would be very useful on it's own, but that could shift the focus out of what is the most important thing. Being able to do that however would lead to some new sound design options - performing FM modulation on a played back sample? Resynthesizing a sample in PADsynth? Well, yes, let's not dive into that or we'll never fix our bugs :D

**Comment by unfa** [comment]
For example I have an EEG print of my heartbeat. I wanted to trace that and use it as a waveform for sound design. If I could load a FLAC file into ZynAddSubFX oscillator module I could do some really neat stuff with this. (I still have to figure out a way to transfer a waveform printed on a piece of paper into a digital audio file...)
